drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,735 @@
+        config1_reg |= implicit << 2;
+        break;
+    default:
+        DEBUG("Unknown device, %d", dev->driver_type);
+        break;
+    }
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, config1_reg);
+}
+
+uint8_t sx127x_get_payload_length(sx127x_t *dev)
+{
+    return dev->settings.lora.payload_len;
+}
+
+void sx127x_set_payload_length(sx127x_t *dev, uint8_t len)
+{
+    if (dev->settings.lora.implicit_header_mode) {
+        dev->settings.lora.payload_len = len;
+        sx127x_reg_write(dev, SX127X_REG_LR_PAYLOADLENGTH, len);
+    }
+}
+
+static inline uint8_t sx127x_get_pa_select(uint32_t channel)
+{
+    if (channel < LORA_RF_MID_BAND_THRESH) {
+        return SX127X_RF_PACONFIG_PASELECT_PABOOST;
+    }
+    else {
+        return SX127X_RF_PACONFIG_PASELECT_RFO;
+    }
+}
+
+uint8_t sx127x_get_power(sx127x_t *dev)
+{
+    return dev->settings.lora.power;
+}
+
+void sx127x_set_power(sx127x_t *dev, uint8_t power)
+{
+    dev->settings.lora.power = power;
+
+    uint8_t pa_config = sx127x_reg_read(dev, SX127X_REG_PACONFIG);
+    uint8_t pa_dac = sx127x_reg_read(dev, SX127X_REG_PADAC);
+
+    pa_config = ((pa_config & SX127X_RF_PACONFIG_PASELECT_MASK) |
+                 sx127x_get_pa_select(dev->settings.channel) << 7);
+    /* max power is 14dBm */
+    pa_config = (pa_config & SX127X_RF_PACONFIG_MAX_POWER_MASK) | (0x05 << 4);
+
+    sx127x_reg_write(dev, SX127X_REG_PARAMP, SX127X_RF_PARAMP_0050_US);
+
+    if ((pa_config & SX127X_RF_PACONFIG_PASELECT_PABOOST)
+        == SX127X_RF_PACONFIG_PASELECT_PABOOST) {
+        if (power > 17) {
+            pa_dac = ((pa_dac & SX127X_RF_PADAC_20DBM_MASK) |
+                      SX127X_RF_PADAC_20DBM_ON);
+        } else {
+            pa_dac = ((pa_dac & SX127X_RF_PADAC_20DBM_MASK) |
+                      SX127X_RF_PADAC_20DBM_OFF);
+        }
+        if ((pa_dac & SX127X_RF_PADAC_20DBM_ON) == SX127X_RF_PADAC_20DBM_ON) {
+            if (power < 5) {
+                power = 5;
+            }
+            if (power > 20) {
+                power = 20;
+            }
+
+            pa_config = ((pa_config & SX127X_RF_PACONFIG_OUTPUTPOWER_MASK) |
+                         (uint8_t)((uint16_t)(power - 5) & 0x0F));
+        } else {
+            if (power < 2) {
+                power = 2;
+            }
+            if (power > 17) {
+                power = 17;
+            }
+
+            pa_config = ((pa_config & SX127X_RF_PACONFIG_OUTPUTPOWER_MASK) |
+                         (uint8_t)((uint16_t)(power - 2) & 0x0F));
+        }
+    } else {
+        if (power < -1) {
+            power = -1;
+        }
+        if (power > 14) {
+            power = 14;
+        }
+
+        pa_config = ((pa_config & SX127X_RF_PACONFIG_OUTPUTPOWER_MASK) |
+                     (uint8_t)((uint16_t)(power + 1) & 0x0F));
+    }
+
+    sx127x_reg_write(dev, SX127X_REG_PACONFIG, pa_config);
+    sx127x_reg_write(dev, SX127X_REG_PADAC, pa_dac);
```

Comment:
`SX127X_REG_PADAC` this register doesn't have the same address for SX1272 & SX1276. 
In the current state, it only works for SX1276